### DISCUSSION TITLE
New RP microcontroller coming

### DISCUSSION THREAD

**Discussion by obdevel** [earlephilhower/arduino-pico#2226] [General]
https://forums.raspberrypi.com/viewtopic.php?t=371165

How might this core accommodate the new part(s) ? I guess it depends on whether the existing C SDK handles both parts, and whether it requires a separate toolchain (if it's not Cortex-M0).

**Comment by earlephilhower** [comment]
If the SDK handles it, then it should be relatively straightforward (famous last words).  AFAIK, the existing GCC can build for all ARM architectures with the proper command line options which can be set per-board.

We'll just have to wait and see what they come out with!

**Comment by Andy2No** [comment]
Exciting!  I hope they fix the ADCs.  That's my only real problem with the RP2040.  They're not just noisy, they're fundamentally flawed, as has been documented.  I'd like to see more ADC inputs too.  Having only 3 is very limiting after using AVR based Arduinos with 6 or more.

**Comment by earlephilhower** [comment]
They've gone public:  https://www.raspberrypi.com/products/rp2350/

Cortex-M33x2, RISC-Vx2, DSP(NEON?) and hardware FPU.  Pretty neat, but not yet available from the usual suspects.

**Reply by earlephilhower** [reply]
Datasheet: https://datasheets.raspberrypi.com/rp2350/rp2350-datasheet.pdf

**Comment by Andy2No** [comment]
Here's the link for the Raspberry Pi Pico 2 board itself, with links to PDFs for product brief and C++/Python SDKs:

https://www.raspberrypi.com/products/raspberry-pi-pico-2/

**Comment by Andy2No** [comment]
This page contains the Pico 2 pinout, as a .svg image and a PDF:

https://www.raspberrypi.com/documentation/microcontrollers/pico-series.html#pinout-and-design-files

Here's the .svg pinout converted to a .gif:

![pico-2-r4-pinout](https://github.com/user-attachments/assets/f444b750-1b27-4080-8ec0-b4c223ea9665)

I see there's still no GP29 pin, for ADC3 but I'm more interested in whether they've improved the noise and linearity of the three it has.  Pimoroni have a 16 pin Tiny series one based on it (Tiny 2350), with all four ADC pins:

https://shop.pimoroni.com/products/tiny-2350?variant=42092638699603

The reduced number of pins would be a real problem for some applications, and it's more expensive by about 50%, but for making something really small, with all four ADC pins, it could be a good option.  They already have those in stock too.  It's nice to see they've stuck with the USB-C socket, as on the Tiny2040, and I'm a little disappointed that the Raspberry Pi Foundation have stuck with the micro USB on the Pico 2.

![tiny2350-1_1500x1500_crop_center](https://github.com/user-attachments/assets/409aa4e1-2446-4b0e-98b8-0a1a5e5121f0)

![tiny2350-2_1500x1500_crop_center](https://github.com/user-attachments/assets/becb808a-136b-445e-bbb9-b1851a970987)

**Comment by obdevel** [comment]
The Tools menu is going to become even more unmanageable ! 

"As for leaping into the world of RISC-V, Upton explained how the cores would work: "They're selectable at boot time: Each port into the bus fabric can be connected either to an M33 or a Hazard3 via a mux. You can even, if you're feeling obtuse, run with one of each."

W version expected 'by the end of the year'.

**Comment by PontusO** [comment]
An arduino-pico port is on it's way (https://github.com/PontusO/arduino-pico-rp2350). Unfortunatly there were a ton of PR's on the pico-sdk the past few days so it is not ready for prime time yet. It works fairly well in its current state though but this is using a 2 month old version of the pico-sdk.

**Reply by earlephilhower** [reply]
Just pushed out the last pre-2.0 SDK/RP2350 support release, so as all those SDK changes come in and break stuff users will have a working older version until all that get cleared out. :smile: 

Looking forward to that massive PR!  They have also updated the original RP2040 (esp. WiFi model) support as I read the changelog so there's a lot of work all around!

**Comment by Andy2No** [comment]
It seems there has been an improvement in the design of the ADC:

```
12.4.1. Changes from RP2040
• Removed spikes in differential nonlinearity at codes 0x200, 0x600, 0xa00 and 0xe00, 
as documented by erratum RP2040-E11, improving the ADC’s precision by around 0.5 ENOB
```

For the problem with the RP2040 ADC, see:

https://www.hackster.io/news/raspberry-pi-confirms-it-is-investigating-a-flaw-in-the-raspberry-pi-pico-rp2040-adc-95c393b55dfb

ADC has very high DNL spikes and low-noise mode ... GitHub › raspberrypi › pico-feedback › issues
https://github.com/raspberrypi/pico-feedback/issues/91

**Comment by XFer012** [comment]
New micro looks nicer and nicer; this post contains interesting stuff, I hope it's OK to link it here
https://dmitry.gr/?r=06.+Thoughts&proj=11.+RP2350

**Comment by obdevel** [comment]
Picked one up on Friday from the RPi retail store here in Cambridge, UK. Rationed to one per customer, although I did say I might come back later wearing a false moustache ;)

As a data point, micropython seems rock solid in both Arm and RISC-V forms. I don't know which version of the SDK this is built from.

**Comment by Hedda** [comment]
Raspberry Pi Pico SDK 2.0.0 link and changelog for reference:

https://github.com/raspberrypi/pico-sdk/releases/tag/2.0.0

https://github.com/raspberrypi/pico-sdk/releases

**Comment by Andy2No** [comment]
It didn't seem appropriate to mention this in the ongoing pull requests (e.g. #2337), but Pimoroni (UK) are showing stock of the new Pico 2 and their 16 pin Tiny 2350 version, with the extra ADC pin:

https://shop.pimoroni.com/collections/pico

They ship worldwide. e.g. by UPS:

https://shop.pimoroni.com/pages/shipping-information

Their main checkout can sometimes refuse to accept Paypal, but the express Paypal checkout link still works, in that case.  I'm not associated with them etc, but I do sometimes buy from them and they have good customer support.

**Comment by obdevel** [comment]
The Pico 2 W has been released. Available from the usual sources.

**Reply by Hedda** [reply]
> The Pico 2 W has been released. Available from the usual sources.

Raspberry Pi Pico 2 W (Wireless model with Wi-Fi and Bluetooth) is on sale now:

- https://www.raspberrypi.com/news/raspberry-pi-pico-2-w-on-sale-now/

Nice specifications, and those specs with the same WiFi and BLE chip will of couse also soon be used to create RP2350 clones and other boards based on RP2350 using this Pico 2 W as reference hardware design:

- https://www.cnx-software.com/2024/11/25/7-raspberry-pi-pico-2-w-board-2-4-ghz-wifi-4-bluetooth-5-2-wireless-module/
